File: langgraph/data/equiv/3.py
from typing import List


def check_negative_balance(transactions: List[int]) -> bool:
    current_balance = 0

    index = 0
    while index < len(transactions):
        current_balance += transactions[index]
        if current_balance < 0:
            return True
        index += 1

    return False

# method name: below_zero -> check_negative_balance
# parameter name: operations -> transactions
# variable name: balance -> current_balance
# changed for loop to while loop with index increment

File: langgraph/result/qwen/32b-eq2js-10/3.js
function checkNegativeBalance(transactions) {
    let currentBalance = 0;

    let index = 0;
    while (index < transactions.length) {
        currentBalance += transactions[index];
        if (currentBalance < 0) {
            return true;
        }
        index += 1;
    }

    return false;
}